### PR TITLE
Improve javadocs

### PR DESCRIPTION
**Pull request by dexgs**
The main change in this PR is adding "package-info.java" files to each package which allows writing a summary (as well as a more detailed explanation) for each package.

This helps to make the generated documentation more navigable:
image

I also improved the javadoc coverage in classes i PR'd earlier (as well as catching a few errors I saw while reading over them).

### CODE REVIEW

**Review by LiamOdero — APPROVED** [review]

**Review by JeremyXie123 — DISMISSED** [review]
Looks good, can't wait to see the fully compiled Javadocs!

**Review by JeremyXie123 — APPROVED** [review]
Documentation is very thorough. I didn't really find anything wrong with it, in fact it's much easier to understand what is gong on. The various optimizations you did like putting the color sliders in a for loop were a great idea.

File: src/main/java/app/media/Media.java [hunk 1]
```diff
@@ -18,6 +18,22 @@
 
 /**
  * Core entity representing a piece of media which can be displayed on a page.
+ *
+ * Media and its sub-classes all have the following properties:
+ * <ul>
```
[the remark this answers is not in the document]
Comment (JeremyXie123) [reply src/main/java/app/media/Media.java:23]: I see.

File: src/main/java/app/media/Media.java [hunk 2]
```diff
@@ -100,27 +144,64 @@ private void setFields(
...
+     *
+     * @throws UnsupportedOperationException if the placeholder ID is assigned,
+     * or if an ID is assigned when the Media already has a non-placeholder ID.
+     */
+    public final void setID(long id) throws UnsupportedOperationException {
+        if (this.id == EMPTY_ID && id != EMPTY_ID) {
             this.id = id;
         } else {
-            throw new Exception("Cannot assign ID to Media twice.");
+            throw new UnsupportedOperationException();
```
[the remark this answers is not in the document]
Comment (JeremyXie123) [reply src/main/java/app/media/Media.java:204]: Ah, that makes sense yeah.

**Review by LiamOdero — APPROVED** [review]

**Review by kaprixal — APPROVED** [review]

**Pull request merged**